statsmodels/stats/contingency_tables.py

```diff
@@ -0,0 +1,1312 @@
+
+        Parameters
+        ----------
+        alpha : float
+            `1 - alpha` is the nominal coverage probability of the confidence
+            intervals.
+        """
+
+        def fmt(x):
+            if type(x) is str:
+                return x
+            return float_format % x
+
+        headers = ["Estimate", "SE", "LCB", "UCB", "p-value"]
+        stubs = ["Odds ratio", "Log odds ratio", "Risk ratio", "Log risk ratio"]
+
+        lcb1, ucb1 = self.oddsratio_confint(alpha)
+        lcb2, ucb2 = self.log_oddsratio_confint(alpha)
+        lcb3, ucb3 = self.riskratio_confint(alpha)
+        lcb4, ucb4 = self.log_riskratio_confint(alpha)
+        data = [[fmt(x) for x in [self.oddsratio, "", lcb1, ucb1, self.oddsratio_pvalue()]],
+                [fmt(x) for x in [self.log_oddsratio, self.log_oddsratio_se, lcb2, ucb2,
+                                  self.oddsratio_pvalue()]],
+                [fmt(x) for x in [self.riskratio, "", lcb2, ucb2, self.riskratio_pvalue()]],
+                [fmt(x) for x in [self.log_riskratio, self.log_riskratio_se, lcb4, ucb4,
+                                  self.riskratio_pvalue()]]]
+        tab = iolib.SimpleTable(data, headers, stubs, data_aligns="r",
+                                table_dec_above='')
+        return tab
+
+
+
+class StratifiedTables(object):
+    """
+    Analyses for a collection of 2x2 stratified contingency tables.
+
+    This class implements the 'Cochran-Mantel-Haenszel' and
+    'Breslow-Day' procedures for analyzing collections of 2x2
+    contingency tables.
+
+    Parameters
+    ----------
+    tables : list
+        A list containing 2x2 contingency tables.
+
+    Attributes
+    ----------
+    logodds_pooled : float
+        An estimate of the pooled log odds ratio.  This is the
+        Mantel-Haenszel estimate of an odds ratio that is common to
+        all the tables.
+    log_oddsratio_se : float
+        The estimated standard error of the pooled log odds ratio,
+        following Robins, Breslow and Greenland (Biometrics
+        42:311-323)
+    oddsratio_pooled : float
+        An estimate of the pooled odds ratio.  This is the
+        Mantel-Haenszel estimate of an odds ratio that is common to
+        all tables.
+    risk_pooled : float
+        An estimate of the pooled risk ratio.  This is an estimate of
+        a risk ratio that is common to all the tables.
+
+    Notes
+    -----
+    This results are based on a sampling model in which the units are
+    independent within and between strata.
+    """
+
+    def __init__(self, tables, shift_zeros=False):
+
+        # Create a data cube
+        tables = [np.asarray(x) for x in tables]
+        table = [x[:, :, None] for x in tables]
+        table = np.concatenate(table, axis=2).astype(np.float64)
+
+        if shift_zeros:
+            zx = (table == 0).sum(0).sum(0)
+            ix = np.flatnonzero(zx > 0)
+            if len(ix) > 0:
+                table[:, :, ix] += 0.5
```

Comment:
this looks the same as

```
if shift_zeros:
    table[table == 0] += 0.5
```

but if table is a view on the original table of the user, then we need a copy if we modify it.